test/extended/router/h2spec.go

```diff
@@ -0,0 +1,286 @@
+		g.It("should pass the h2spec conformance tests", func() {
+			err := oc.Run("create").Args("-f", configPath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			e2e.ExpectNoError(oc.KubeFramework().WaitForPodRunning("h2spec-nginx"))
+
+			routeTypeTests := []h2specRouteTypeTest{
+				{
+					routeType: "edge",
+					knownFailures: map[string]bool{
+						"generic/3.1.1": true,
+						"http2/4.2.3":   true,
+					},
+				},
+				{
+					routeType: "passthrough",
+					knownFailures: map[string]bool{
+						"hpack/4.2.1":     true,
+						"hpack/5.2.1":     true,
+						"http2/4.2.2":     true,
+						"http2/5.1.1":     true,
+						"http2/5.1.11":    true,
+						"http2/5.1.2":     true,
+						"http2/5.1.3":     true,
+						"http2/5.1.5":     true,
+						"http2/5.1.8":     true,
+						"http2/5.4.1.1":   true,
+						"http2/5.4.1.2":   true,
+						"http2/6.1.1":     true,
+						"http2/6.1.2":     true,
+						"http2/6.4.2":     true,
+						"http2/6.5.2":     true,
+						"http2/6.7.3":     true,
+						"http2/6.8.1":     true,
+						"http2/6.9.1.3":   true,
+						"http2/8.1.2.1.1": true,
+						"http2/8.1.2.1.2": true,
+						"http2/8.1.2.1.4": true,
+						"http2/8.1.2.2.1": true,
+						"http2/8.1.2.2.2": true,
+						"http2/8.1.2.3.1": true,
+						"http2/8.1.2.3.2": true,
+						"http2/8.1.2.3.3": true,
+						"http2/8.1.2.3.4": true,
+						"http2/8.1.2.3.5": true,
+						"http2/8.1.2.3.6": true,
+						"http2/8.1.2.3.7": true,
+						"http2/8.1.2.6.1": true,
+						"http2/8.1.2.6.2": true,
+					},
+				},
+				{
+					routeType: "reencrypt",
+					knownFailures: map[string]bool{
+						"http2/4.2.3": true,
+					},
+				},
+			}
+
+			for _, t := range routeTypeTests {
+				if t.routeType == "edge" {
+					e2e.Logf("Skipping %q tests; waiting for: https://github.com/openshift/router/pull/104", t.routeType)
+					continue
+				}
+				if t.routeType == "passthrough" {
+					// TODO(frobware): work out why the underlying nginx config errors so much
+					e2e.Logf("Skipping %q tests", t.routeType)
+					continue
+				}
+
+				t.hostname = getHostnameForRoute(oc, fmt.Sprintf("h2spec-nginx-%s", t.routeType))
+				g.By(fmt.Sprintf("[%s] Running conformance tests against host %q", t.routeType, t.hostname))
+
+				// sampleForFailures when true runs
+				// the conformance tests for N
+				// iterations to identify flaking
+				// tests. This should only be set to
+				// true in development (i.e., not CI).
+				//
+				// This should be run every time we
+				// consume any new version of haproxy
+				// be that a major, minor or a micro
+				// update to continuously validate the
+				// set of test case IDs that fail.
+				if _, sampleForFailures := os.LookupEnv("ROUTER_H2SPEC_SAMPLE_MODE"); sampleForFailures {
+					runConformanceTestsAndLogAggregateFailures(oc, t, 150)
+					continue
+				}
+
+				testSuites := runConformanceTests(oc, t)
+				o.Expect(len(testSuites)).ShouldNot(o.BeZero(), "expected > 0 testSuites")
```

Comment:
Ginkgo-golf:

```
o.Expect(testSuites).ShouldNot(o.BeEmpty())
```

Reply:
Addressed in https://github.com/openshift/origin/pull/24674/commits/798acd38e4abc3439c5e905a8a090afd838ec3b5